packages/esm-patient-flags-app/src/hooks/usePatientFlags.tsx

```diff
@@ -0,0 +1,51 @@
+import { openmrsFetch } from '@openmrs/esm-framework';
+import { useEffect, useMemo, useState } from 'react';
+import useSWR from 'swr';
+
+interface FlagFetchResponse {
+  uuid: string;
+  name: string;
+  message: string;
+}
+
+interface FlagsFetchResponse {
+  results: Array<FlagFetchResponse>;
+}
+
+/**
+ * React hook that takes in a patient uuid and returns
+ * patient flags for that patient together with helper objects
+ * @param patientUuid Unique patient idenfier
+ * @returns An array of patient identifiers
+ */
+export function useFlagsFromPatient(patientUuid: string) {
+  const { data, error, isValidating, mutate } = useSWR<{ data: FlagsFetchResponse }, Error>(
+    `/ws/rest/v1/patientflags/flag?patient=${patientUuid}&v=full`,
+    openmrsFetch,
+  );
+  const result = useMemo(() => {
+    return {
+      flags: data?.data?.results ?? null,
+      flagsError: error,
+      isflagsLoading: !data && !error,
+      isflagsValidating: isValidating,
+      mutate,
+    };
+  }, [data, error, isValidating, mutate]);
+  return result;
+}
+
+export function useCurrentPath(): string {
```

Comment:
Why not just wrap the component in React Router and use the existing `useLocation` hook?

Reply:
I had issues reusing the `useLocation` hook.  Calling it threw an uncaught Error which i think might have been caused by the `patient-dashboad` not being a child of `<Route>`.

Binding my component into `<Route>` also didn't solve this problem hence I just decided to create my own function watching every change to `window.location.pathname`